clients/vault/src/horizon/client.rs

```diff
@@ -0,0 +1,254 @@
+		let vault_address = vault_keypair.get_public();
+
+		let request_url = String::from(horizon_url(is_public_network)) +
+			"/accounts/" + str::from_utf8(vault_address.to_encoding().as_slice())
+			.map_err(|_| Error::HttpFetchingError)? +
+			"/transactions?order=desc&limit=1";
+
+		println!("request url: {:?}", request_url);
+
+		let horizon_response = self.client.get_transactions(request_url.as_str()).await;
+
+		horizon_response
+	}
+
+	fn is_unhandled_transaction(&mut self, tx: &Transaction) -> bool {
+		const UP_TO_DATE: () = ();
+		let latest_tx_id_utf8 = &tx.id;
+
+		let prev_tx_id = &self.last_tx_id;
+		let initial = !matches!(prev_tx_id, Some(_));
+
+		let result = match prev_tx_id {
+			Some(prev_tx_id) =>
+				if prev_tx_id == latest_tx_id_utf8 {
+					Err(UP_TO_DATE)
+				} else {
+					Ok(latest_tx_id_utf8.clone())
+				},
+			None => Ok(latest_tx_id_utf8.clone()),
+		};
+
+		match result {
+			Ok(latest_tx_id) => {
+				self.last_tx_id = Some(latest_tx_id.clone());
+				if !initial {
+					println!(
+                        "Found new transaction from Horizon (id {:#?}). Starting to process new transaction",
+                        str::from_utf8(&latest_tx_id).unwrap()
+                    );
+					tracing::info!(
+                        "Found new transaction from Horizon (id {:#?}). Starting to process new transaction",
+                        str::from_utf8(&latest_tx_id).unwrap()
+                    );
+
+					true
+				} else {
+					println!("Initial transaction handled");
+					tracing::info!("Initial transaction handled");
+					false
+				}
+			},
+			Err(UP_TO_DATE) => {
+				println!("Already up to date");
+				tracing::info!("Already up to date");
+				false
+			},
+		}
+	}
+
+	async fn fetch_horizon_and_process_new_transactions(
+		&mut self,
+		issue_set: Arc<Mutex<IssueRequests>>,
+		handler: Arc<Mutex<ScpMessageHandler>>,
+	) {
+		let handler = handler.lock().await;
+
+		let res = self.fetch_latest_txs(handler.is_public_network).await;
+		let transactions = match res {
+			Ok(txs) => txs._embedded.records,
+			Err(e) => {
+				tracing::warn!("Failed to fetch transactions: {:?}", e);
+				println!("oh no, :( failed to fetch transactions: {:?}", e);
+				return
+			},
+		};
+
+		if transactions.len() > 0 {
+			let tx = transactions[0].clone();
+			let id = tx.id.clone();
+			if self.is_unhandled_transaction(&tx) && is_tx_relevant(&tx, issue_set.clone()).await {
+				match handler.watch_slot(tx.ledger.try_into().unwrap()).await {
+					Ok(_) => {
+						println!("following transaction {:?}", String::from_utf8(id.clone()));
+						tracing::info!("following transaction {:?}", String::from_utf8(id));
+					},
+					Err(e) => {
+						println!("Failed to watch transaction: {:?}", e);
+						tracing::error!("Failed to watch transaction: {:?}", e);
+					},
+				}
+			}
+		}
+	}
+}
+
+pub async fn poll_horizon_for_new_transactions(
+	vault_secret_key: String,
+	issue_set: Arc<Mutex<IssueRequests>>,
```

Comment:
Nit: Using an [RwLock](https://docs.rs/tokio/0.2.22/tokio/sync/struct.RwLock.html) might be more elegant because it allows distinguishing between read and write access.